src/databricks/labs/dqx/check_funcs.py

```diff
@@ -19,13 +19,101 @@
 )
 
 _IPV4_OCTET = r"(25[0-5]|2[0-4]\d|1\d{2}|[1-9]?\d)"
+_IPV6_HEXTET = r"([0-9a-fA-F]{1,4})"
+_IPV6_UNCOMPRESSED = rf"^{_IPV6_HEXTET}(:{_IPV6_HEXTET}){{7}}$"
+_IPV6_COMPRESSED = (
+    r"^("
+    + r"::("
+    + _IPV6_HEXTET
+    + r":){0,6}"
+    + _IPV6_HEXTET
+    + r"?|"
+    + r"("
+    + _IPV6_HEXTET
+    + r":){1}:("
+    + _IPV6_HEXTET
+    + r":){0,5}"
+    + _IPV6_HEXTET
+    + r"?|"
+    + r"("
+    + _IPV6_HEXTET
+    + r":){2}:("
+    + _IPV6_HEXTET
+    + r":){0,4}"
+    + _IPV6_HEXTET
+    + r"?|"
+    + r"("
+    + _IPV6_HEXTET
+    + r":){3}:("
+    + _IPV6_HEXTET
+    + r":){0,3}"
+    + _IPV6_HEXTET
+    + r"?|"
```

Comment:
@cornzyblack i extended the pattern as it was not catching all cases. I also added more cases to the tests.